ibis/impala/compiler.py

```diff
@@ -631,10 +632,18 @@ def _string_literal_format(translator, expr):
 
 def _number_literal_format(translator, expr):
     value = expr.op().value
-    formatted = repr(value)
 
-    if formatted in {'nan', 'inf', '-inf'}:
-        return "CAST({!r} AS DOUBLE)".format(formatted)
+    if math.isfinite(value):
+        formatted = repr(value)
+    else:
+        if math.isnan(value):
+            formatted_val = "'NaN'"
+        elif math.isinf(value):
+            if value > 0:
+                formatted_val = "'Infinity'"
+            else:
+                formatted_val = "'-Infinity'"
+        formatted = "CAST({} AS DOUBLE)".format(formatted_val)
```

Comment:
Can you use `"CAST({!r} AS DOUBLE)".format(fomatted_val)` and remove the embedded single quotes from nan, infinity, etc?